ionoscloud/resource_backup_unit.go

```diff
@@ -346,3 +347,33 @@ func setBackupUnitData(d *schema.ResourceData, backupUnit *ionoscloud.BackupUnit
 	}
 	return nil
 }
+
+// BackupUnitFindByID it's a custom function that returns a backup unit using its ID. The difference
+// between this function and the client BackupUnitsApi.BackupunitsFindById is that this function
+// retrieves the whole list of backup units and then iterates over them, returning the backup unit
+// when the ID is matched. This function was implemented because the response values in
+// the "BackupUnitsApi.BackupunitsFindById" method are invalid, but the values in
+// "BackupUnitsApi.BackupunitsGet" method are valid. This is a workaround for the API problem with
+// invalid values in the response for BackupUnitsApi.BackupunitsFindById.
+func BackupUnitFindByID(ctx context.Context, backupUnitID string, client *ionoscloud.APIClient) (ionoscloud.BackupUnit, *ionoscloud.APIResponse, error) {
+	backupUnits, apiResponse, err := client.BackupUnitsApi.BackupunitsGet(ctx).Depth(2).Execute()
+	var backupUnit ionoscloud.BackupUnit
+	logApiRequestTime(apiResponse)
+	if err != nil {
+		return backupUnit, apiResponse, fmt.Errorf("error while retrieving the list of backup units: %w", err)
+	}
+	if backupUnits.Items == nil {
+		return backupUnit, apiResponse, fmt.Errorf("expected a list of backup units in the response but received 'nil' instead")
+	}
+	for _, backupUnit := range *backupUnits.Items {
+		if backupUnit.Id == nil {
+			return backupUnit, apiResponse, fmt.Errorf("expected a backup unit with a valid ID but received 'nil' instead")
+		}
+		if *backupUnit.Id == backupUnitID {
+			return backupUnit, apiResponse, nil
+		}
+	}
+	// If the backup unit wasn't found, simulate a 404 error.
+	apiResponse.StatusCode = http.StatusNotFound
```

Comment:
it's weird to mutate the API Response's status code. The rest of the API Response will not reflect a 404 error. For example the Status string, etc.

This object will now have fields populated like an OK response, except the StatusCode will be 404

Reply:
@avirtopeanu-ionos `it's weird to mutate the API Response's status code.` I agree, I made this change since the status code it's the only thing we look at when deciding if the backup unit it's still there. I'm open to suggestions to avoid this weird mutation.

Reply:
we either mutate the apiresponse, or we check the error with error.is.

Reply:
let's add a todo in the function comment that we need to replace the function once the api fixes the findByID.  This being a temporary fix, it doesn;t matter

Reply:
@cristiGuranIonos done